libqpdf/qpdf-c.cc

```diff
@@ -1238,6 +1238,22 @@ QPDF_BOOL qpdf_oh_get_bool_value(qpdf_data qpdf, qpdf_oh oh)
         });
 }
 
+QPDF_BOOL qpdf_oh_get_value_as_bool(
+    qpdf_data qpdf, qpdf_oh oh, QPDF_BOOL* value)
+{
+    return do_with_oh<QPDF_BOOL>(
+        qpdf, oh, return_false, [value](QPDFObjectHandle& o) {
+            QTC::TC("qpdf", "qpdf-c called qpdf_oh_get_value_as_bool");
+            bool v = *value;
+            QPDF_BOOL result = o.getValueAsBool(v);
+            if (result)
+            {
+                *value = QPDF_TRUE;
```

Comment:
This doesn't look right. Why not just `o.getValueAsBool(*value)`? I don't see how this would ever set *value to false.

Reply:
or `*value = v` or `*value = v ? QPDF_TRUE : QPDF_FALSE`

Reply:
Changed . 
*value does not work as a bool&